x/leveragelp/keeper/utils.go

```diff
@@ -62,3 +64,30 @@ func (k Keeper) GetAmmPool(ctx sdk.Context, poolId uint64) (ammtypes.Pool, error
 	}
 	return ammPool, nil
 }
+
+func (k Keeper) GetLeverageLpUpdatedLeverage(ctx sdk.Context, positions []*types.Position) ([]*types.QueryPosition, error) {
+	updatedLeveragePositions := []*types.QueryPosition{}
+	for i, position := range positions {
+		baseCurrency, found := k.assetProfileKeeper.GetUsdcDenom(ctx)
+		if !found {
+			return nil, errorsmod.Wrapf(assetprofiletypes.ErrAssetProfileNotFound, "asset %s not found", ptypes.BaseCurrency)
+		}
+		exitCoinsAfterFee, _, err := k.amm.ExitPoolEst(ctx, position.GetAmmPoolId(), position.LeveragedLpAmount, baseCurrency)
+		if err != nil {
+			return nil, err
+		}
+
+		exitAmountAfterFee := exitCoinsAfterFee.AmountOf(baseCurrency)
+
+		price := k.oracleKeeper.GetAssetPriceFromDenom(ctx, position.Collateral.Denom)
+		updated_leverage := exitAmountAfterFee.Quo(exitAmountAfterFee.Sub(position.Liabilities.Mul(price.TruncateInt())))
```

Comment:
```suggestion
		updated_leverage := exitAmountAfterFee.Quo(exitAmountAfterFee.Sub(position.Liabilities))
```

Reply:
I don't think we need to multiply with usdc price here because, it will cancel out.

exitAmountAfterFee is in USDC
liability is also in USDC

Reply:
yes